### PR TITLE
SIARDDK support

### PR DESCRIPTION
**Pull request by AntonioG70**

### CODE REVIEW

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/input/SIARDDK1007ImportModule.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  public DatabaseImportModule getDatabaseImportModule() {
+    final Map<String, String> properties = MapUtils.buildMapFromObjects(SIARDDK1007ModuleFactory.PARAMETER_FOLDER,
+      mainContainer.getPath().normalize().toAbsolutePath().toString(), SIARDDK1007ModuleFactory.PARAMETER_AS_SCHEMA,
+      paramImportAsSchema);
+    return new SIARDImportDefault(moduleName, contentStrategy, mainContainer, readStrategy, metadataStrategy,
+      properties);
+  }
+
+}
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/input/SIARDDK1007ImportModule.java:70]: Please avoid code repetition

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK1007MetadataImportStrategy.java [hunk 2]
```diff
@@ -0,0 +1,478 @@
+/**
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK1007MetadataImportStrategy.java:1]: Please avoid code repetition

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+      }
+    }
+
+    return virtualForeignKeys;
+  }
+
+  private TableStructure createVirtualTable() throws ModuleException {
+    try {
+      DocIndexType docIndexType = loadVirtualTableMetadata();
+      TableStructure virtualTable = new TableStructure();
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java:359]: Create a new VirtualTableStructure class that extends TableStructure, in the constructor of this class configure the object with everything that a virtual table needs to have by default, such as name, description, etc.

With this you will be able to differentiate a table from a virtual table using intanceof, which can be useful for applications that use dbptk-developer as a library.

I will review the rest of the class after this change

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+      virtualTable.setIndex(currentTableIndex++);
+      virtualTable.setSchema(getImportAsSchemaName());
+      virtualTable.setId(String.format("%s.%s", virtualTable.getSchema(), "virtual_table"));
+      virtualTable.setName("virtual_table");
+      virtualTable.setDescription("A virtual table");
+      virtualTable.setRows(docIndexType.getDoc().size());
+      virtualTable.setColumns(createVirtualTableColumns());
+      virtualTable.setPrimaryKey(createVirtualPrimaryKey("dID"));
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java:367]: Do the same for columns, primary keys, foreign keys and anything else that is virtual.

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/typeConverter/SQLStandardDatatypeImporter.java [hunk 5]
```diff
@@ -163,8 +163,13 @@ protected Type getNationalCharType(String typeName, int columnSize, int decimalD
...
     SimpleTypeString type = new SimpleTypeString(columnSize, false);
-    type.setSql99TypeName("CHARACTER", columnSize);
-    type.setSql2008TypeName("CHARACTER", columnSize);
+    if (columnSize == 0) {
+      type.setSql99TypeName("CHARACTER");
+      type.setSql2008TypeName("CHARACTER");
+    } else {
+      type.setSql99TypeName("CHARACTER", columnSize);
+      type.setSql2008TypeName("CHARACTER", columnSize);
+    }
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/typeConverter/SQLStandardDatatypeImporter.java:172]: This does not seem to be part of this PR, please create a new PR and address which issue this modification belongs to.

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/ArchiveFileIndexInputStreamStrategy.java [hunk 6]
```diff
@@ -29,10 +29,19 @@ class ArchiveFileIndexInputStreamStrategy implements FileIndexXsdInputStreamStra
...
+    String xsdFilePath = siarddk1007PathImportStrategy.getXsdFilePath(SIARDDKConstants.FILE_INDEX);
+
+    return readStrategy.createInputStream(mainFolder, xsdFilePath);
+  }
+
+  @Override
+  public InputStream getInputStream(SIARDDK128PathImportStrategy siarddk128PathImportStrategy) throws ModuleException {
+    ReadStrategy readStrategy = siarddk128PathImportStrategy.getReadStrategy();
+    SIARDArchiveContainer mainFolder = siarddk128PathImportStrategy.getMainFolder();
+    String xsdFilePath = siarddk128PathImportStrategy.getXsdFilePath(SIARDDKConstants.FILE_INDEX);
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/ArchiveFileIndexInputStreamStrategy.java:44]: After you change the code so that you don't have duplicate code, here you can possibly use polymorphism and continue with just one method.

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/SIARDDK1007PathImportStrategy.java [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+    return readStrategy;
+  }
+
+  /**
+   * @return the mainFolder
+   */
+  public SIARDArchiveContainer getMainFolder() {
+    return mainFolder;
+  }
+}
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/SIARDDK1007PathImportStrategy.java:330]: Please avoid code repetition

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/SIARDDK128PathImportStrategy.java [hunk 8]
```diff
@@ -65,11 +62,12 @@ public class SIARDDKPathImportStrategy implements ContentPathImportStrategy, Met
...
   // required for fileIndex.xml in the standard
   protected byte[] tabelIndexExpectedMD5Sum;
+  protected byte[] archiveIndexExpectedMD5Sum;
   protected boolean fileIndexIsParsed;
 
-  public SIARDDKPathImportStrategy(SIARDArchiveContainer mainFolder, ReadStrategy readStrategy,
-    MetadataPathStrategy metadataPathStrategy, String importAsSchema,
-    FileIndexXsdInputStreamStrategy fileIndexXsdInputStreamStrategy) {
+  public SIARDDK128PathImportStrategy(SIARDArchiveContainer mainFolder, ReadStrategy readStrategy,
+                                      MetadataPathStrategy metadataPathStrategy, String importAsSchema,
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/SIARDDK128PathImportStrategy.java:69]: Please format your code

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/content/SIARDDK1007ContentExportStrategy.java [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+
+  private String encodeText(String s) {
+    s = s.trim();
+    s = s.replace("<", "&lt;");
+    s = s.replace(">", "&gt;");
+    s = s.replace("&", "&amp;");
+    s = s.replace("'", "&apos;");
+    s = s.replace("\"", "&quot;");
+    return s;
+  }
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/content/SIARDDK1007ContentExportStrategy.java:415]: Please avoid code repetition

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/metadata/SIARDDK1007ContextDocumentationWriter.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
+/**
```
Comment (006627) [inline dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/metadata/SIARDDK1007ContextDocumentationWriter.java:1]: Please avoid code repetition

**Review by 006627 — CHANGES_REQUESTED** [review]
A lot of code is repeated due to the SIARD DK version separation, please review and apply best practices to ensure good code quality, I will finish reviewing this PR when you have these changes ready.

File: dbptk-core/src/test/java/com/databasepreservation/siarddk/SIARDDKTestUtil.java [hunk 11]
```diff
@@ -37,8 +37,8 @@ static void assertArchiveFoldersEqualAfterExportImport(Path archiveFldToProcessP
       FileUtils.deleteDirectory(archFile);
     }
 
-    String[] argumentsToMain = new String[] {"migrate", "--import=siard-dk", "--import-as-schema=public", "--import-folder",
-      archiveFldToProcessPath.toString(), "--export", "siard-dk", "--export-folder",
+    String[] argumentsToMain = new String[] {"migrate", "--import=siard-dk-1007", "--import-as-schema=public", "--import-folder",
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-core/src/test/java/com/databasepreservation/siarddk/SIARDDKTestUtil.java:40]: I ran the same command without applying any changes to the code and the result seems correct.
`      Import modules: 
        import-config
        jdbc
        microsoft-access
        microsoft-sql-server
        mysql
        oracle
        postgresql
        progress-openedge
        siard-1
        siard-2
        siard-dk-1007
        siard-dk-128
        sybase


      Export modules: 
        import-config
        jdbc
        microsoft-sql-server
        mysql
        oracle
        postgresql
        siard-1
        siard-2
        siard-dk-1007
        siard-dk-128
`

File: dbptk-core/src/test/resources/siarddk/AVID.SA.18001.1/Indices/fileIndex.xml [hunk 12]
```diff
@@ -35,11 +35,6 @@
     <fiN>1.tif</fiN>
     <md5>3EC3F367397407EF0E53642E3F1FFD21</md5>
   </f>
-  <f>
-    <foN>AVID.SA.18001.1\Indices</foN>
-    <fiN>archiveIndex.xml</fiN>
-    <md5>16EFBE43D5F278162428A625FF0CA18C</md5>
-  </f>
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-core/src/test/resources/siarddk/AVID.SA.18001.1/Indices/fileIndex.xml:42]: It was a mistake, i will put that back.

File: dbptk-core/src/test/java/com/databasepreservation/testing/integration/siard/SiardTest.java [hunk 13]
```diff
@@ -153,7 +154,21 @@ public void SIARD_RoundTrip(SiardVersion version)
         }
       }
     }
-
+    for (SchemaStructure orgSchema : other.getSchemas()) {
+      for (TableStructure tableStructure : other.getSchemas().get(0).getTables()) {
+        if (tableStructure.getName().equals("virtual_table")) {
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-core/src/test/java/com/databasepreservation/testing/integration/siard/SiardTest.java]: done

File: dbptk-core/src/test/java/com/databasepreservation/testing/integration/siard/SiardTest.java [hunk 14]
```diff
@@ -153,7 +154,21 @@ public void SIARD_RoundTrip(SiardVersion version)
...
+          other.getSchemas().get(0).getTables().remove(tableStructure);
+        }
+      }
+    }
+    for (SchemaStructure schemaStructure : other.getSchemas()) {
+      for (TableStructure tableStructure : schemaStructure.getTables()) {
+        List<ForeignKey> foreignKeys = tableStructure.getForeignKeys();
+        if (foreignKeys != null) {
+          foreignKeys.removeIf(fk -> fk.getName().equals("FK_virtual_table"));
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-core/src/test/java/com/databasepreservation/testing/integration/siard/SiardTest.java]: done

File: dbptk-model/src/main/java/com/databasepreservation/SIARDEdition.java [hunk 15]
```diff
@@ -177,6 +177,8 @@ private static HashMap<Parameter, String> buildImportParameters(HashMap<String,
 
       if (key != null && key.longName().contentEquals("file")) {
         importParameters.put(key, entry.getValue().get(0));
+      } else if (key != null && key.longName().contentEquals("folder")) {
+        importParameters.put(key, entry.getValue().get(0));
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-model/src/main/java/com/databasepreservation/SIARDEdition.java:181]: siarddk has "folder" as its parameter instead of "file"

File: dbptk-model/src/main/java/com/databasepreservation/model/structure/DatabaseStructure.java [hunk 16]
```diff
@@ -1002,7 +1012,8 @@ public boolean equals(Object obj) {
       if (other.url != null) {
         return false;
       }
-    } else if (!url.equals(other.url)) {
+    }
+    else if (!url.equals(other.url)) {
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-model/src/main/java/com/databasepreservation/model/structure/DatabaseStructure.java]: done

File: dbptk-model/src/main/java/com/databasepreservation/utils/XMLUtils.java [hunk 17]
```diff
@@ -7,11 +7,14 @@
...
 package com.databasepreservation.utils;
 
+import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.StringWriter;
 import java.io.Writer;
+import java.util.ArrayList;
 import java.util.Iterator;
+import java.util.List;
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-model/src/main/java/com/databasepreservation/utils/XMLUtils.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007EditFactory.java [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Miguel Guimarães <mguimaraes@keep.pt>
+ */
+public class SIARDDK1007EditFactory implements EditModuleFactory {
+
+  public static final String PARAMETER_FOLDER = "folder";
+  // public static final String PARAMETER_SET_REGEX = "regex"; -- IDEA
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007EditFactory.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007EditFactory.java [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Miguel Guimarães <mguimaraes@keep.pt>
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007EditFactory.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007EditFactory.java [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+   */
+
+  @Override
+  public String getModuleName() {
+    return "edit-siard-dk-1007";
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return false;
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007EditFactory.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007ModuleFactory.java [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+import com.databasepreservation.model.parameters.Parameter;
+import com.databasepreservation.model.parameters.Parameter.CATEGORY_TYPE;
+import com.databasepreservation.model.parameters.Parameter.INPUT_TYPE;
+import com.databasepreservation.model.parameters.Parameters;
+import com.databasepreservation.modules.siard.constants.SIARDDKConstants;
+import com.databasepreservation.modules.siard.out.output.SIARDDK1007ExportModule;
+
+/**
+ * @author Andreas Kring <andreas@magenta.dk>
+ * @author Thomas Kristensen <tk@bithuset.dk>
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK1007ModuleFactory.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK128EditFactory.java [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+    if (Files.notExists(pFolder)) {
+      throw new SiardNotFoundException().withPath(pFolder.toAbsolutePath().toString())
+        .withMessage("The path to the siard folder appears to be incorrect");
+    }
+
+    reporter.importModuleParameters(getModuleName(), PARAMETER_FOLDER, pFolder.normalize().toAbsolutePath().toString());
+
+    return new SIARDEditModule(pFolder, SIARDConstants.SiardVersion.DK_128);
+  }
+}
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK128EditFactory.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/common/path/SIARDDK1007MetadataPathStrategy.java [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+      SIARDDKConstants.CONTEXT_DOCUMENTATION_INDEX, SIARDDKConstants.FILE_INDEX,
+      SIARDDKConstants.DOCUMENT_IDENTIFICATION, SIARDDKConstants.XML_SCHEMA, "fileIndex_original", "docIndex_original"};
+
+    ArrayList<String> validFilenames = new ArrayList<String>(Arrays.asList(validFileNames));
+    if (validFilenames.contains(filename)) {
+      return true;
+    }
+    return false;
+  }
+}
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/common/path/SIARDDK1007MetadataPathStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/content/SIARDDK1007ContentImportStrategy.java [hunk 24]
```diff
@@ -0,0 +1,406 @@
...
+    }
+
+  }
+
+  @Override
+  public void characters(char[] ch, int start, int length) throws SAXException {
+    currentTagContentStrBld.append(ch, start, length);
+  }
+
+}
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/content/SIARDDK1007ContentImportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/input/SIARDDK1007ImportModule.java (hunk 1 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/input/SIARDDK1007ImportModule.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK128ModuleFactory.java [hunk 25]
```diff
@@ -7,35 +7,34 @@
...
+import java.util.List;
+import java.util.Map;
 
 /**
  * @author Andreas Kring <andreas@magenta.dk>
  * @author Thomas Kristensen <tk@bithuset.dk>
  *
  */
-public class SIARDDKModuleFactory implements DatabaseModuleFactory {
+public class SIARDDK128ModuleFactory implements DatabaseModuleFactory {
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/SIARDDK128ModuleFactory.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/content/SIARDDK128ContentImportStrategy.java [hunk 26]
```diff
@@ -90,8 +103,8 @@ public class SIARDDKContentImportStrategy extends DefaultHandler implements Cont
    * @author Thomas Kristensen <tk@bithuset.dk>
    *
    */
-  public SIARDDKContentImportStrategy(FolderReadStrategyMD5Sum readStrategy, SIARDDKPathImportStrategy pathStrategy,
-    String importAsSchema) {
+  public SIARDDK128ContentImportStrategy(FolderReadStrategyMD5Sum readStrategy, SIARDDK128PathImportStrategy pathStrategy,
+                                         String importAsSchema) {
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/content/SIARDDK128ContentImportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/content/SIARDDK128ContentImportStrategy.java [hunk 27]
```diff
@@ -121,83 +134,168 @@ public void importContent(DatabaseExportModule dbExportHandler, SIARDArchiveCont
...
-          logger.error("Error validating schema", e);
-          throw new ModuleException().withMessage("Error reading XSD file: "
-            + pathStrategy.getTableXSDFilePath(schema.getName(), table.getId()) + " for table:" + table.getId())
-            .withCause(e);
-        }
-        DigestInputStream currentTableInputStream = readStrategy.createInputStream(currentFolder, xmlFileName,
-          pathStrategy.getTableXMLFileMD5(schema.getName(), table.getId()));
-
-        SAXErrorHandler saxErrorHandler = new SAXErrorHandler();
+        if (!table.getId().split("\\.")[1].equals("virtual_table")) {
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/content/SIARDDK128ContentImportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java (hunk 3 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/typeConverter/SQLStandardDatatypeImporter.java (hunk 5 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/typeConverter/SQLStandardDatatypeImporter.java:172]: Issue created https://github.com/keeps/dbptk-developer/issues/627

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java (hunk 4 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/ArchiveFileIndexInputStreamStrategy.java (hunk 6 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/ArchiveFileIndexInputStreamStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/SIARDDK1007PathImportStrategy.java (hunk 7 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/SIARDDK1007PathImportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/content/SIARDDK1007ContentExportStrategy.java (hunk 9 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/content/SIARDDK1007ContentExportStrategy.java]: done

File: dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/metadata/SIARDDK1007ContextDocumentationWriter.java (hunk 10 above)
Comment (AntonioG70) [reply dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/out/metadata/SIARDDK1007ContextDocumentationWriter.java]: done

**Review by 006627 — APPROVED** [review]
